### PR TITLE
Migrate PR builders from Jenkins to Github Actions

### PR DESCRIPTION
**Pull request by HyukjinKwon**

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -10,20 +10,30 @@ on:
...
+          - 'catalyst,sql'
+          - 'hive'
+          - 'hive-thriftserver'
+          - 'streaming,sql-kafka-0-10,streaming-kafka-0-10'
+          - 'mllib-local,mllib'
+          - 'pyspark-core,pyspark-sql,pyspark-resource,pyspark-streaming,pyspark-mllib,pyspark-ml'
+          - 'sparkr'
+          - 'yarn,mesos,kubernetes,hadoop-cloud,spark-ganglia-lgpl'
+    env:
+      TEST_ONLY_MODULES: ${{ matrix.python-modules }}
```
Comment (gengliangwang) [inline .github/workflows/master.yml:36]: python-modules => modules?

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -57,31 +82,66 @@ jobs:
...
+        restore-keys: |
+          ${{ matrix.java }}-${{ matrix.hadoop }}-ivy2-
     - name: Set up JDK ${{ matrix.java }}
       uses: actions/setup-java@v1
       with:
         java-version: ${{ matrix.java }}
-    - name: Build with Maven
+    # PySpark
+    - uses: actions/setup-python@v2
+      if: contains(matrix.modules, 'pyspark')
```
Comment (gengliangwang) [inline .github/workflows/master.yml:97]: Will this branch be reached? The module names are "pyspark-..."

Comment (HyukjinKwon) [reply .github/workflows/master.yml:97]: This one:

https://github.com/HyukjinKwon/spark/pull/4/files/7f4a4c0b87445204343c901b7c3d6dd55e2fac34#diff-04eb107ee163a50b61281ca08f4e4c7bR41-R43

It checks if the string contains `pyspark`

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -9,148 +9,205 @@ on:
...
-        hive: [ 'hive-1.2', 'hive-2.3' ]
-        exclude:
-        - java: '11'
-          hive: 'hive-1.2'
-        - hadoop: 'hadoop-3.2'
-          hive: 'hive-1.2'
-    name: Build Spark - JDK${{ matrix.java }}/${{ matrix.hadoop }}/${{ matrix.hive }}
-
+        java:
+          - 1.8
```
Comment (dongjoon-hyun) [inline .github/workflows/master.yml:20]: BTW, please don't forget to recover the matrix when you make a PR to Apache Spark community~

Comment (HyukjinKwon) [reply .github/workflows/master.yml:20]: No worry :-)

Comment (HyukjinKwon) [reply .github/workflows/master.yml:20]: No worries :-)

**Pull request closed without merging**